tools/autograd/gen_autograd_functions.py

```diff
@@ -146,6 +148,11 @@ def save_arg(arg, is_output):
             unpack.append('auto {} = {}_.unpack({});'.format(name, name, ptr))
         elif arg['type'] == 'TensorList':
             saved_variables.append('std::vector<SavedVariable> {}_;'.format(name))
+            release_variables.append('for (auto& sv : {}_)'.format(name))
+            release_variables.append('{')
+            release_variables.append('  sv.reset_data();'.format(name))
+            release_variables.append('  sv.reset_grad_function();'.format(name))
+            release_variables.append('}')
             release_variables.append('{}_.clear();'.format(name))
```

Comment:
Another potential solution could be: instead of calling `.clear()` on the vector<SavedVariable> we could  just call `reset_data` and `reset_grad_function` on each element. Then we wouldn't need to have an extra `released` flag and autograd errors out as usual when one tries to unpack an undefined tensor from the SavedVariable

This might end up using more memory, though.

Reply:
Edit: nevermind, this doesn't work. If one has an empty TensorList then the problem still happens.